src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+			}
+
+			if (getDockToRectFromTargetElement)
+			{
+				var targetTopLeft = default(Point);
+				Target.TransformToVisual(null).TransformPoint(targetTopLeft);
+
+				//// UNO TODO
+				//      // targetTopLeft.X should be the left edge of the target, so adjust for RTL by
+				//      // subtracting the width of the target.
+				//      if (bIsRTL)
+				//      {
+				//          targetTopLeft.X -= targetActualWidth;
+				//      }
+
+				rcDockTo.X = targetTopLeft.X;
+				rcDockTo.Y = targetTopLeft.Y;
+				rcDockTo.Width = Target.ActualWidth;
+				rcDockTo.Height = Target.ActualHeight;
+			}
+
+			rcDockTo = rcDockTo.OffsetRect(origin.X, origin.Y);
+
+			// If horizontal & vertical offset are not specified, use the system defaults.
+			var isPropertyLocal = this.IsDependencyPropertySet(HorizontalOffsetProperty);
+			if (!isPropertyLocal)
+			{
+				// UNO TODO
+				//switch (m_inputMode)
+				//{
+				//case AutomaticToolTipInputMode::Keyboard:
+				//    horizontalOffset = DEFAULT_KEYBOARD_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Mouse:
+				//    horizontalOffset = DEFAULT_MOUSE_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Touch:
+				//    horizontalOffset = DEFAULT_TOUCH_OFFSET;
+				//    break;
+				//}
+				horizontalOffset = DEFAULT_MOUSE_OFFSET;
+			}
+
+			isPropertyLocal = this.IsDependencyPropertySet(VerticalOffsetProperty);
+			if (!isPropertyLocal)
+			{
+				// UNO TODO
+				//switch (m_inputMode)
+				//{
+				//case AutomaticToolTipInputMode::Keyboard:
+				//    verticalOffset = DEFAULT_KEYBOARD_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Mouse:
+				//    verticalOffset = DEFAULT_MOUSE_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Touch:
+				//    verticalOffset = DEFAULT_TOUCH_OFFSET;
+				//    break;
+				//}
+
+				verticalOffset = DEFAULT_MOUSE_OFFSET;
+			}
+
+			// UNO TODO
+			//// Reverse Left/Right palacement for RTL, because the target is flipped.
+			//if (bIsRTL)
+			//{
+			//    if (placement == PlacementMode.Right)
+			//    {
+			//        placement = PlacementMode.Left;
+			//    }
+			//    else if (placement == PlacementMode.Left)
+			//    {
+			//        placement = PlacementMode.Right;
+			//    }
+			//}
+
+			// To honor horizontal/vertical offset, inflate the placement target by these offsets before calling into
+			// the Windows popup positioning logic.
+			rcDockTo.Inflate(horizontalOffset, verticalOffset);
+			(var rcResult, var placementChosen) = ToolTipPositioning.QueryRelativePosition(
+				constraint,
+				szFlyout,
+				rcDockTo,
+				placement);
+
+			// if right/bottom doesn't fit into the plug-in bounds, clip the ToolTip
+			{
+				var clipCalculationsRect = new Rect(0, 0, ActualWidth, ActualHeight);
+				CalculateTooltipClip(clipCalculationsRect, visibleRect.Width - visibleRect.X, visibleRect.Height - visibleRect.Y);
+			}
+
+			Console.WriteLine($"wjkreujk rcResult={rcResult}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)